### PR TITLE
Added mailchimp 

### PR DESCRIPTION
**Pull request by mdesmet**

### CODE REVIEW

**Comment by mdesmet** [comment]
This is the API part for https://github.com/DivanteLtd/vue-storefront/issues/1899

**Comment by mdesmet** [comment]
I've made the API only return the status instead of leaking mailchimp internals. The goal is to be able to adapt to other mail list operators (I'm planning to also provide implementation for Magento 2).

File: src/api/extensions/mailchimp-subscribe/index.js [hunk 1]
```diff
@@ -16,25 +42,24 @@ module.exports = ({ config, db }) => {
...
+      json: true,
+      body: { members: [ { email_address: userData.email, status: 'subscribed' } ], "update_existing": true }
+    }, function (error, response, body) {
+      if (error) {
+        console.error(error)
+        apiStatus(res, 'An error occured while accessing Mailchimp', 500)
+      } else {
+        apiStatus(res, body.status, 200)
+      }
+    })
```
Comment (pkarw) [inline src/api/extensions/mailchimp-subscribe/index.js]: This is duplicate of the busines logic between `/get` and `/post` moreover not sure why we modify data on `/get` - shouldn't be only one method supproted: `post` ?

Comment (mdesmet) [reply src/api/extensions/mailchimp-subscribe/index.js]: I agree the response handler can be generalized. 

The get doesn't modify data on Mailchimp, it also doesn't leak any Mailchimp data, except for the subscriber status to make the API more general and able to be swapped out to for example the magento 2 api.

What do you mean by only one method should be supported?

GET-> retrieve the subscriber status
POST -> subscribe
DELETE -> unsubscribe

Comment (pkarw) [reply src/api/extensions/mailchimp-subscribe/index.js]: OK now it makes perfect sense, ignore my comment

File: src/api/extensions/mailchimp-subscribe/index.js [hunk 2]
```diff
@@ -1,13 +1,39 @@
...
-	
 
-	/** 
-	 * POST create an user
+  /**
+   * Retrieve user subscription status
+   */
+  mcApi.get('/subscribe', (req, res) => {
+
+    let email = req.query.email
```
Comment (lukeromanowicz) [inline src/api/extensions/mailchimp-subscribe/index.js]: @mdesmet this should be a const

**Comment by lukeromanowicz** [comment]
Waiting for merging #2754

**Pull request merged**